packages/playwright-test/src/runner.ts

```diff
@@ -385,6 +411,7 @@ export class Runner {
       for (const [project, files] of filesByProject) {
         const grepMatcher = createTitleMatcher(project.grep);
         const grepInvertMatcher = project.grepInvert ? createTitleMatcher(project.grepInvert) : null;
+        const groupTitleMatcher = phase.find(p => p.projectName.toLocaleLowerCase() === project.name.toLocaleLowerCase())!.testTitleMatcher;
```

Comment:
Let's explore having a class for RunPhase or something like that?

Reply:
I'll do it in a separate PR as discussed offline.